sf-fx-runtime-java-runtime/src/main/java/com/salesforce/functions/jvm/runtime/util/ClassLoaderUtils.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+package com.salesforce.functions.jvm.runtime.util;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Optional;
+
+public class ClassLoaderUtils {
+
+  /**
+   * Copies a file from a ClassLoader to disk.
+   *
+   * <p>This exists since URLClassLoader does not work well with "jar:" URLs. As a workaround, we
+   * copy the JAR file from the JAR file to a temporary location and load the classes from there.
```

Comment:
Is "copy the JAR file from the JAR file" a typo?

Reply:
Nope. We actually have JAR files in JAR files.

![](https://media.giphy.com/media/9UqRcQHzBou6A/giphy.gif)